### PR TITLE
Added layouts and automatic file extension

### PR DESCRIPTION
**Pull request by berarma**
Added a method to use layout template files. Also, the .php extension in template file names is optional. The trailing slash in the template path is optional too. And some CS fixes.

### CODE REVIEW

File: src/PhpRenderer.php [hunk 1]
```diff
@@ -45,30 +70,37 @@ public function __construct($templatePath = "")
      *
      * @return ResponseInterface
      *
-     * @throws \InvalidArgumentException
      * @throws \RuntimeException
      */
     public function render(ResponseInterface $response, $template, array $data = [])
     {
-        if (isset($data['template'])) {
-            throw new \InvalidArgumentException("Duplicate template key found");
```
Comment (geggleto) [inline src/PhpRenderer.php]: You must include this type-check regardless of any kind of cuteness.

File: src/PhpRenderer.php [hunk 2]
```diff
@@ -23,16 +23,41 @@ class PhpRenderer
...
+    protected $layout;
+
+    /**
      * SlimRenderer constructor.
      *
      * @param string $templatePath
      */
     public function __construct($templatePath = "")
     {
+        if ($templatePath !== '') {
```
Comment (geggleto) [inline src/PhpRenderer.php]: add layout to the constructor as optional

**Comment by berarma** [comment]
Done.

**Comment by geggleto** [comment]
//cc @akrabat looks good to me

@berarma  As a separate PR would you mind updating the readme.md ?

**Comment by berarma** [comment]
OK, I'll do. I'm new to Slim and just starting my first project with it. If you can think of a better way to do it please tell me.

**Comment by geggleto** [comment]
I no longer think that this will be required. Please see #19 
In 2.1 you will be able to render sub-templates in the view.

**Comment by geggleto** [comment]
Closing In favour of #19 , #21

**Pull request closed without merging**